Xamarin.PropertyEditing/IObjectEditor.cs

```diff
@@ -51,9 +52,27 @@ public interface IObjectEditor
 		/// </summary>
 		event EventHandler<EditorPropertyChangedEventArgs> PropertyChanged;
 
-		void SetValue<T> (IPropertyInfo property, ValueInfo<T> value, PropertyVariation variation = null);
+		/*
+		 * Dealing with async values in the context of what's possible across all target platforms is a bit complex.
+		 * While implicit safety may be able to be ensured, it would be exhaustive to reason it out and could change
+		 * at any moment, so we need an explicit safety strategy. As we make value gets and sets async, we open ourselves
+		 * to race conditions even on just the UI thread. Value changes can come from the user, or from the object editor
+		 * at any point. So, we break these changes into two categories: Blocking UI & Waiting Values.
+		 *
+		 * Anything that needs to deal with the current state of the values will Wait asynchronously on prior changes to
+		 * finish. These largely involve value changes from the object editor, which just acts as a INPC and does not
+		 * contain the value itself, so it will be safe and the latest value when queried upon reaching active state in
+		 * the wait queue (see AsyncWorkQueue). Selected objects (when >1) need to wait on values as they need to be
+		 * able to calculate the intersection values.
+		 *
+		 * Anything that needs to deal with user input will need to block its UI. We simply can't reason about the user
+		 * changing values while waiting for previous changes, because they value they're changing may change underneath
+		 * them or become invalid as a result of a prior pending change.
+		 */
+
+		Task SetValueAsync<T> (IPropertyInfo property, ValueInfo<T> value, PropertyVariation variation = null);
 
 		/// <exception cref="ArgumentNullException"><paramref name="property"/> is <c>null</c>.</exception>
-		ValueInfo<T> GetValue<T> (IPropertyInfo property, PropertyVariation variation = null);
+		Task<ValueInfo<T>> GetValueAsync<T> (IPropertyInfo property, PropertyVariation variation = null);
```

Comment:
Bringing it back from offline discussion: I don't really think we buy ourselves enough to add it and implicitly encourage people to implement cancellation.